ria_remote/create_sibling_ria.py

```diff
@@ -0,0 +1,391 @@
+        recursive=recursion_flag,
+        recursion_limit=recursion_limit,
+    )
+
+    @staticmethod
+    @datasetmethod(name='create_sibling_ria')
+    @eval_results
+    def __call__(url,
+                 name,
+                 dataset=None,
+                 ria_remote=None,
+                 force=False,
+                 post_update_hook=False,
+                 shared=None,
+                 group=None,
+                 no_ria_remote=False,
+                 recursive=False,
+                 recursion_limit=None
+                 ):
+
+        # TODO: is check_installed actually required?
+        ds = require_dataset(dataset, check_installed=True, purpose='create sibling RIA')
+        res_kwargs = dict(
+            ds=ds,
+            action="create-sibling-ria",
+            logger=lgr,
+        )
+
+        if ds.repo.get_hexsha() is None or ds.id is None:
+            raise RuntimeError(
+                "Repository at {} is not a DataLad dataset, "
+                "run 'datalad create' first.".format(ds.path))
+
+        if no_ria_remote and ria_remote:
+            raise ValueError("no-ria-remote and ria-remote were given simultanously.")
+
+        if not no_ria_remote and not ria_remote:
+            ria_remote = "{}-ria".format(name)
+
+        if not no_ria_remote and name == ria_remote:
+            # leads to unresolvable, circular dependency with publish-depends
+            raise ValueError("sibling names must not be equal")
+
+        # TODO: messages - this is "create-sibling". Don't confuse existence of local remotes with existence of the
+        #       actual remote sibling in wording
+        ds_siblings = [r['name'] for r in ds.siblings(result_renderer=None)]
+        if not force and name in ds_siblings:
+            yield get_status_dict(
+                status='error',
+                message="a sibling '{}' is already configured. Use --force to overwrite it.".format(name),
+                **res_kwargs,
+            )
+            return
+
+        if not no_ria_remote and not force and ria_remote in ds_siblings:
+            yield get_status_dict(
+                status='error',
+                message="a storage-sibling '{}' is already configured. Use --force to overwrite it.".format(ria_remote),
+                **res_kwargs,
+            )
+            return
+
+        # parse target URL
+        # Note: For python API we should be able to deal with RI and its subclasses. However, currently quite a dance
+        # needed (see below)
+        src_url_ri = RI(url) if not isinstance(url, RI) else url
+        if src_url_ri.fragment:
+            # TODO: This is still somewhat fragile. If the given dataset id in fragment matches, it's actually fine.
+            #       But then: What if additional things are given (like @somebranch)?
+            #       However, ideally we expect no fragment at all.
+            lgr.warning("Ignoring unexpected URL fragment '%s'." % src_url_ri.fragment)
```

Comment:
See your point, but in those other cases we just pass it on and let fail wherever this would rightfully lead to failure. `fragment` on the other hand we are actively writing to, thereby replacing any existing one.

Reply:
Sorry, I wasn't precise. There should be no warning, this is an error. We expect a store URL nothing else. If it has a fragment, or anything else, it is not we expect, and likely the user is not aware of what we expect. Not being clever mean fail IMHO.

Reply:
Okay